### PR TITLE
New resource `azurerm_data_factory_linked_service_odbc`

### PR DESCRIPTION
**Pull request by onrylmz**
Fixes [14770](https://github.com/hashicorp/terraform-provider-azurerm/issues/14770)

### CODE REVIEW

File: website/docs/r/data_factory_linked_service_odbc.html.markdown [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
+
+* `create` - (Defaults to 30 minutes) Used when creating the Data Factory Odbc Linked Service.
+* `update` - (Defaults to 30 minutes) Used when updating the Data Factory Odbc Linked Service.
+* `read` - (Defaults to 5 minutes) Used when retrieving the Data Factory Odbc Linked Service.
+* `delete` - (Defaults to 30 minutes) Used when deleting the Data Factory Odbc Linked Service.
+
+## Import
+
+Data Factory Odbc Linked Service's can be imported using the `resource id`, e.g.
```
Comment (stephybun) [inline website/docs/r/data_factory_linked_service_odbc.html.markdown:103]:
```suggestion
* `username` - (Required) The username which can be used to authenticate to the ODBC endpoint.

* `password` - (Required) The password associated with the username, which can be used to authenticate to the ODBC endpoint.

## Attributes Reference

The following attributes are exported:

* `id` - The ID of the Data Factory ODBC Linked Service.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Data Factory ODBC Linked Service.
* `update` - (Defaults to 30 minutes) Used when updating the Data Factory ODBC Linked Service.
* `read` - (Defaults to 5 minutes) Used when retrieving the Data Factory ODBC Linked Service.
* `delete` - (Defaults to 30 minutes) Used when deleting the Data Factory ODBC Linked Service.

## Import

Data Factory ODBC Linked Services can be imported using the `resource id`, e.g.
```

File: internal/services/datafactory/data_factory_linked_service_odbc_resource.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceArmDataFactoryLinkedServiceOdbc() *pluginsdk.Resource {
```
Comment (stephybun) [inline internal/services/datafactory/data_factory_linked_service_odbc_resource.go:19]: Could we also remove `Arm` from all of the function names for consistency?
```suggestion
func resourceDataFactoryLinkedServiceOdbc() *pluginsdk.Resource {
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @onrylmz! Overall this is looking good. Once the comments have been addressed and the test failure is fixed we can have another look over this 🙂 

```
------- Stdout: -------
=== RUN   TestAccDataFactoryLinkedServiceOdbc_basic_auth
=== PAUSE TestAccDataFactoryLinkedServiceOdbc_basic_auth
=== CONT  TestAccDataFactoryLinkedServiceOdbc_basic_auth
testcase.go:110: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
(map[string]string) {
}
(map[string]string) (len=4) {
(string) (len=22) "basic_authentication.#": (string) (len=1) "1",
(string) (len=24) "basic_authentication.0.%": (string) (len=1) "2",
(string) (len=31) "basic_authentication.0.password": (string) (len=9) "Ch4ngeM3!",
(string) (len=31) "basic_authentication.0.username": (string) (len=4) "emma"
}
--- FAIL: TestAccDataFactoryLinkedServiceOdbc_basic_auth (174.26s)
FAIL
```

**Comment by onrylmz** [comment]
Thanks for the quick reply @stephybun, hope I did address all the missing parts.

```
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryLinkedServiceODBC_basic_auth -timeout 60m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataFactoryLinkedServiceODBC_basic_auth
=== PAUSE TestAccDataFactoryLinkedServiceODBC_basic_auth
=== CONT  TestAccDataFactoryLinkedServiceODBC_basic_auth
--- PASS: TestAccDataFactoryLinkedServiceODBC_basic_auth (97.02s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   98.886s
```

**Review by stephybun — APPROVED** [review]
Thanks for incorporating those changes, LGTM 👍🏼

**Pull request merged**